core/src/language/mod.rs

```diff
@@ -302,4 +395,97 @@ pub trait Language {
 
         Ok(())
     }
+
+    /// Types that are remapped will be excluded from import references.
+    fn ignored_reference_types(&self) -> Vec<String> {
+        Vec::new()
+    }
+
+    /// Any other final steps after modules have been generated. For example creating a new
+    /// module with special types.
+    fn post_generation(&self, _output_folder: &str) -> Result<(), GenerationError> {
+        Ok(())
+    }
+}
+
+/// Lookup any refeferences to other typeshared types in order to build
+/// a list of imports for the generated module.
+fn used_imports<'a, 'b: 'a>(
+    data: &'b ParsedData,
+    all_types: &'a CrateTypes,
+) -> ScopedCrateTypes<'a> {
+    let mut used_imports = BTreeMap::new();
+
+    // If we have reference that is a re-export we can attempt to find it with the
+    // following heuristic.
+    let fallback = |referenced_import: &'a ImportedType, used: &mut ScopedCrateTypes<'a>| {
+        // Find the first type that does not belong to the current crate.
+        if let Some((crate_name, ty)) = all_types
+            .iter()
+            .flat_map(|(k, v)| {
+                v.iter()
+                    .find(|&t| t == &referenced_import.type_name && k != &data.crate_name)
+                    .map(|t| (k, t))
+            })
+            .next()
+        {
+            println!("Warning: Using {crate_name} as module for {ty} which is not in referenced crate {}", referenced_import.base_crate);
+            used.entry(crate_name)
+                .and_modify(|v| {
+                    v.insert(ty.as_str());
+                })
+                .or_insert(BTreeSet::from([ty.as_str()]));
+        } else {
+            // println!("Could not lookup reference {referenced_import:?}");
+        }
+    };
+
+    for referenced_import in data
+        .import_types
+        .iter()
+        // Skip over imports that reference the current crate. They
+        // are all collapsed into one module per crate.
+        .filter(|imp| imp.base_crate != data.crate_name)
+    {
+        // Look up the types for the referenced imported crate.
+        if let Some(type_names) = all_types.get(&referenced_import.base_crate) {
+            // We can have "*" wildcard here. We need to add all.
+            if referenced_import.type_name == "*" {
+                used_imports
+                    .entry(&referenced_import.base_crate)
+                    .and_modify(|names: &mut BTreeSet<&str>| {
+                        names.extend(type_names.iter().map(|s| s.as_str()))
+                    });
+                continue;
+            }
+
+            // Add referenced import for each matching type.
+            if let Some(ty_name) = type_names.get(&referenced_import.type_name) {
```

Comment:
nit: Normally I like continue but here I think `else if`  better communicates the intent

Reply:
Updated